src/goto-symex/symex_assign.cpp

```diff
@@ -250,7 +250,8 @@ static assignmentt rewrite_with_to_field_symbols(
           update.new_value().type());
       }
 
-      state.field_sensitivity.apply(ns, state, field_sensitive_lhs, true);
+      field_sensitive_lhs = state.field_sensitivity.apply(
```

Comment:
!!! Test for this fix? Also mention in the commit message that this is actually a bugfix (or fix it in a separate commit?)